jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/diagram/ProcessDiagramPresenter.java

```diff
@@ -61,29 +60,27 @@ public void onProcessInstanceSelectionEvent(@Observes ProcessInstanceSelectionEv
         String serverTemplateId = event.getServerTemplateId();
 
         if (processInstanceId != null && !processInstanceId.isEmpty()) {
-            processImageService.call(new RemoteCallback<String>() {
-                @Override
-                public void callback(final String svgContent) {
-                    designerWidget.displayImage(svgContent);
-                }
-            }).getProcessInstanceDiagram(serverTemplateId,
-                                         containerId,
-                                         Long.parseLong(processInstanceId));
+            processImageService.call((String svgContent) -> displayImage(svgContent)).getProcessInstanceDiagram(serverTemplateId,
+                                                                                                                containerId,
+                                                                                                                Long.parseLong(processInstanceId));
         }
     }
 
     public void onProcessSelectionEvent(@Observes final ProcessDefSelectionEvent event) {
         String containerId = event.getDeploymentId();
         String serverTemplateId = event.getServerTemplateId();
         String processId = event.getProcessId();
-        processImageService.call(new RemoteCallback<String>() {
-            @Override
-            public void callback(final String svgContent) {
-                designerWidget.displayImage(svgContent);
-            }
-        }).getProcessDiagram(serverTemplateId,
-                             containerId,
-                             processId);
+        processImageService.call((String svgContent) -> displayImage(svgContent)).getProcessDiagram(serverTemplateId,
+                                                                                                    containerId,
+                                                                                                    processId);
+    }
+
+    protected void displayImage(final String svgContent) {
+        if (svgContent == null || svgContent.isEmpty()) {
+            designerWidget.displayMessage(constants.Process_Diagram_Not_Found());
+        } else {
+            designerWidget.displayImage(svgContent);
```

Comment:
Can we add a test or two for this logic?

Reply:
done